### PR TITLE
access request from view + sessions refactor

### PR DESCRIPTION
**Pull request by tanner0101**
This pull request adds access to the `Request` (including sessions, storage, and more) from views. It also greatly simplifies sessions.

### Access Request from View

```swift
drop.get("view") { req in
    return try drop.view.make("form", [
        "name": "Foo"
    ], for: request)
}
```

```leaf
<h1>My Form: #(name)</h1>

#if(request.session.data.user) {
    Welcome #(request.session.data.user.name)
}
```

### Simplified Sessions

The `Session` class itself has been simplified and decoupled from the SessionsProtocol. This allows for easier testing as test Sessions can be created without having to create test SessionProtocols and Middleware.

### CODE REVIEW

**Comment by Casperhr** [comment]
Perfect, that will solve all these issues.

File: Sources/Sessions/CacheSessions.swift [hunk 1]
```diff
@@ -8,18 +8,23 @@ public final class CacheSessions: SessionsProtocol {
...
     }
 
-    public func get(for identifier: String) throws -> Node? {
-        return try cache.get(identifier)
-    }
-
-    public func set(_ value: Node?, for identifier: String) throws {
-        if let value = value {
-            try cache.set(identifier, value)
+    public func get(identifier: String) throws -> Session? {
```
Comment (LoganWright) [inline Sources/Sessions/CacheSessions.swift]: I think `rm` on the original function breaks api

File: Sources/Sessions/MemorySessions.swift [hunk 2]
```diff
@@ -18,38 +18,38 @@ public class MemorySessions: SessionsProtocol {
...
 
-        return value
+        return session
     }
 
     /**
         Sets value for session id at given key
     */
-    public func set(_ value: Node?, for identifier: String) {
+    public func set(_ session: Session) {
```
Comment (LoganWright) [inline Sources/Sessions/MemorySessions.swift]: same thing here, we might need to make a duplicate class or something

File: Sources/Sessions/Session.swift [hunk 3]
```diff
@@ -9,60 +9,16 @@ import Node
     `drop.sessions`.
 */
 public final class Session {
-    public var identifier: String?
-
-    private var sessions: SessionsProtocol
-
-    init(sessions: SessionsProtocol) {
```
Comment (LoganWright) [inline Sources/Sessions/Session.swift]: we might need to maintain this one too.

**Comment by tannernelson** [comment]
I can't think of a good way to deprecate the old Sessions API. Probably best to wait for 2.0 for that. I've created a new PR with just the view/request stuff. https://github.com/vapor/vapor/pull/752

**Pull request closed without merging**